pkg/cloud/services/compute/bastion.go

```diff
@@ -26,35 +26,35 @@ import (
 	capoerrors "sigs.k8s.io/cluster-api-provider-openstack/pkg/utils/errors"
 )
 
-func (s *Service) DeleteBastion(openStackCluster *infrav1.OpenStackCluster, serverID string) error {
-	instance, err := s.GetInstance(serverID)
+func (s *Service) DeleteBastion(openStackCluster *infrav1.OpenStackCluster, clusterName string) error {
+	instance, err := s.InstanceExists(fmt.Sprintf("%s-bastion", clusterName))
 	if err != nil {
 		return err
 	}
-	if instance == nil {
-		return nil
-	}
-	if err = deleteInstance(s, instance.ID); err != nil {
-		record.Warnf(openStackCluster, "FailedDeleteServer", "Failed to delete server %s with id %s: %v", instance.Name, instance.ID, err)
-		return err
-	}
 
```

Comment:
I think we should use a guard clause here:
```go
if instance == nil || instance.ID == "" {
  return nil
}
```

Reply:
Fixed.